api/server/call_logs.go

```diff
@@ -18,24 +18,29 @@ type callLogResponse struct {
 	Log     *models.CallLog `json:"log"`
 }
 
-func writeJSON(c *gin.Context, callID, appName string, logReader io.Reader) {
+func writeJSON(c *gin.Context, callID, appID string, logReader io.Reader) {
 	var b bytes.Buffer
 	b.ReadFrom(logReader)
 	c.JSON(http.StatusOK, callLogResponse{"Successfully loaded log",
 		&models.CallLog{
-			CallID:  callID,
-			AppName: appName,
-			Log:     b.String(),
+			CallID: callID,
+			AppID:  appID,
+			Log:    b.String(),
 		}})
 }
 
 func (s *Server) handleCallLogGet(c *gin.Context) {
 	ctx := c.Request.Context()
 
-	appName := c.MustGet(api.AppName).(string)
+	app, err := s.datastore.GetApp(ctx, &models.App{Name: c.MustGet(api.App).(string)})
```

Comment:
yea we shouldn't have to do stuff like this, it should be in front of every `/v1/apps/:app` endpoint if it exists at all. stuff like this can easily be done in some of the gin middleware like things we have now such as https://github.com/fnproject/fn/blob/master/api/server/gin_middlewares.go#L72 -- where if the request url has that variable in it, we would look up the app by name there, shove it onto an additional key in the request config, and here we can look it up using `MustGet` 

I see a bunch of `GetApp` calls in various front end methods that should be replaced with this kind of behavior.

Reply:
@denismakogon need to do this one too imo